crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -466,10 +471,13 @@ impl SlidingSync {
 
         Span::current().record("pos", &pos);
 
+        // Configure the timeout.
+        let timeout = if all_lists_are_fully_loaded { Some(self.inner.poll_timeout) } else { None };
```

Comment:
Can you expand on that comment, to explain why we don't set the `poll_timeout` sometimes? (basically repeating the commit's message in fewer words, ideally)